lib/modules/manager/custom/jsonata/utils.ts

```diff
@@ -0,0 +1,108 @@
+import is from '@sindresorhus/is';
+import jsonata from 'jsonata';
+import { logger } from '../../../../logger';
+import * as template from '../../../../util/template';
+import { parseUrl } from '../../../../util/url';
+import type { PackageDependency } from '../../types';
+import type { ValidMatchFields } from '../utils';
+import { checkIsValidDependency, validMatchFields } from '../utils';
+import { QueryResultZodSchema } from './schema';
+import type { JSONataManagerTemplates, JsonataExtractConfig } from './types';
+
+export async function handleMatching(
+  json: unknown,
+  packageFile: string,
+  config: JsonataExtractConfig,
+): Promise<PackageDependency[]> {
+  let results: Record<string, string>[] = [];
+  const { matchStrings: jsonataQueries } = config;
+  for (const query of jsonataQueries) {
+    // won't fail as this is verified during config validation
+    const jsonataExpression = jsonata(query);
+    // this does not throw error, just returns undefined if no matches
+    let queryResult = await jsonataExpression.evaluate(json);
+
+    if (
+      !queryResult ||
+      is.emptyObject(queryResult) ||
```

Comment:
do we really need to check it here? wouldn't it fail later anyway 🤔

Reply:
An early return with a some logs gives better visibility of this issue, hence I added it.

Reply:
But, it should allow empty objects to pass, as someone could've configured templates for all the fields ie. currentValue, datasource and packageName.